ui/src/components/GridButton.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState, useCallback } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+
+const GridButton = (i: number) => {
+    const [open, setOpen] = useState(false);
+
+    const handleOpen = useCallback(() => setOpen(true), []);
+    
+    const handleClose = useCallback(() => setOpen(false), []);
+
+    return(
+      <>
+        <Button onClick={handleOpen} key={i}></Button>
```

Comment:
keys are always set in the context they are used in, i.e. When calling the component in the map function. You never pass them in.

https://react.dev/learn/rendering-lists